### PR TITLE
add default constructors in some accessorClasses.cpp classes for empty cases.

### PR DESCRIPTION
**Pull request by perrydv**
Fixes Issue #379

### CODE REVIEW

File: packages/nimble/inst/include/nimble/accessorClasses.h [hunk 1]
```diff
@@ -161,7 +161,8 @@ class SingleVariableMapAccessBase {
  public:
   int offset, length;
   bool singleton;
-  vector<int> sizes, strides; 
+  vector<int> sizes, strides;
+ SingleVariableMapAccessBase() : offset(0), singleton(false) {length = 0;}; /* length(0) triggers the Rf_length macro mixup */
```
Comment (fritzo) [inline packages/nimble/inst/include/nimble/accessorClasses.h]: Remove trailing semicolon here and everywhere else.

Also consider simplifying to `... : offset(0), length(0), singleton(false) {}`

File: packages/nimble/inst/include/nimble/accessorClasses.h [hunk 2]
```diff
@@ -285,6 +287,7 @@ class SingleVariableAccessBase {
  public:
   int flatIndexStart, flatIndexEnd; // For 1-3: In R these should be 1,3; In C++: 0, 3
   int length; // copying function can check for singletons if it wants to
+ SingleVariableAccessBase() : flatIndexStart(0), flatIndexEnd(0) { length = 0; };
```
Comment (fritzo) [inline packages/nimble/inst/include/nimble/accessorClasses.h]: simplify to `... : flatIndexStart(0), flatIndexEnd(0), length(0) {}`

**Comment by danielturek** [comment]
If it fixes the problem, then looks good to me.  Thanks @perrydv

File: packages/nimble/inst/include/nimble/accessorClasses.h (hunk 1 above)
Comment (perrydv) [inline packages/nimble/inst/include/nimble/accessorClasses.h]: length(0) triggers error from R's macro `length` which replaces `length(x)` with `Rf_length(x)`.  You're getting us fixed on that but meanwhile `length(0)` breaks.

**Comment by perrydv** [comment]
I cleaned up the trailing semi-colons.  That is such a trivial change I am now going to squash and merge. @danielturek this should let you move ahead with Issue #379 fixed

**Comment by danielturek** [comment]
Thanks!

**Pull request merged**